src/ray/gcs/gcs_server/test/gcs_actor_manager_test.cc

```diff
@@ -1236,8 +1236,12 @@ TEST_F(GcsActorManagerTest, TestGetAllActorInfoFilters) {
     rpc::GetAllActorInfoRequest request;
     request.mutable_filters()->set_actor_id(actor->GetActorID().Binary());
 
+#if GOOGLE_PROTOBUF_VERSION < 5029000
```

Comment:
I'm not sure why we need the macro, don't we have single version of protobuf?
https://github.com/ray-project/ray/blob/e393a716d8742a987a36df555defb2ca90bb94d4/bazel/ray_deps_setup.bzl#L90-L102

Reply:
IMO, it's highly useful for ones who build Ray from sources (like me) to be able to use dependency versions newer than ones predefined by Bazel.

Reply:
I'm not sure what do you mean here. Let me try this way, as long a you build with bazel and `bazel build`, you will fixiate on one particular version.

Reply:
Except you 
- Have another protobuf target, that's what [brazil](https://gist.github.com/terabyte/15a2d3d407285b8b5a0a7964dd6283b0) but not bazel
- Or you're willing to build on your own build system (i.e. make, cmake, system dependency), which is highly discouraged

Reply:
> be able to use dependency versions newer than ones predefined by Bazel.

Do you have an example command/script on how you plan to build your own?

Reply:
@dentiny Later it will be C++20/23 and Flatbuffers :) I was working on associated PRs in parallel to this one.

IMO, the burden is much smaller than benefits it brings - ability to use latest library versions if needed. 

In theory, this PR can be reworked to upgrade Ray in general to use the latest dependency versions.

Reply:
> In my case, I'm forcing it to use already built Protobuf v5.29.0-dev instead of the version hardcoded in .bzl file.

Are you talking about prebuilt ELF?

What if I want to patch protobuf (i.e. for debugging and profiling)?
How do you plan to deal with different compilation/link options?

Reply:
> Later it will be C++20/23 and Flatbuffers :)

I'm not sure I understand, C++ version is fixed in toolchain also, which should be one single version.

Reply:
A high-level question, do you have a proposal doc?
I think this PR needs more discussion, a few things:
- What's the benefit of having multiple version of protobuf in our system, while bazel encourages (1) checking from source code and single version;
- For multiple versions, how you plan to maintain the compatibility? What's a proper multi-version usage, what's not;
- For pre-built ELF, how do you plan to deal with different compilation options and patches.

Reply:
> In my case, I'm forcing it to use already built Protobuf v5.29.0-dev instead of the version hardcoded in .bzl file.

Sure technically you could do that, but do we need to checkin the ELF into production code?

Reply:
@dentiny I better understand your position now, thanks for clarifying! I believe now, you are right in that we shouldn't support multiple versions. I'll take some time to rework this PR into one updating the Protobuf version in Bazel configs and associated code.

Per C++, the same, yeah. I'll think how do it better. 

Per ELF question, not sure if I understood your correctly here. If you've meant pre-built Protobuf binary on Linux, it was my particular use case of building Ray with deps built outside the Bazel in favor of artifacts reusability. Neither this nor planned PRs encourage that as it may be too complex for general use-case.

Reply:
> Per ELF question, not sure if I understood your correctly here. If you've meant pre-built Protobuf binary on Linux, it was my particular use case of building Ray with deps built outside the Bazel in favor of artifacts reusability.

I don't favor using ELF as dependency in our codebase, though you can.
- System dependency is discouraged, it breaks [hermeticity](https://bazel.build/basics/hermeticity) 
- Self-built ELF is also not good, since it loses the capability to build with different options

Reply:
If you're working on particular hardware, you should use toolchain for cross-compilation: https://bazel.build/concepts/platforms

Reply:
@dentiny We don't use cross-compilation as building on the same hardware where resultant binaries will run. However, we use Bazel toolchains to define the presets of appropriate optimizations flags like `-mcpu=neoverse-v2`.